pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -31,20 +31,40 @@ func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
 		return fmt.Errorf("cluster %s not adopted. No namespace in the clusterdocument", mon.oc.Name)
 	}
 
+	return mon.validateHiveConditions(ctx)
+}
+
+func (mon *Monitor) validateHiveConditions(ctx context.Context) error {
 	cd, err := mon.hiveclientset.HiveV1().ClusterDeployments(mon.oc.Properties.HiveProfile.Namespace).Get(ctx, hive.ClusterDeploymentName, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
+	conditions := mon.filterConditions(ctx, cd, clusterDeploymentConditionsExpected)
+
+	mon.emitMetrics(conditions)
+
+	return nil
+}
+
+func (mon *Monitor) filterConditions(ctx context.Context, cd *hivev1.ClusterDeployment, clusterDeploymentConditionsExpected map[hivev1.ClusterDeploymentConditionType]corev1.ConditionStatus) []hivev1.ClusterDeploymentCondition {
```

Comment:
Name is too generic to be in `Monitor`: monitor works with a bunch of other types which have conditions.

Reply:
Is specific to ClusterDeployments now